docs/adr/adr_backend_etl_and_hosting_for_StatViz.md

```diff
@@ -0,0 +1,237 @@
+3. **URL Filters**: Filters, like which organization or base a visualization targets, can be passed as dimensions or as GraphQL query variables.
+
+#### A.3 N-to-N dimensions
+
+Handling many-to-many (N-to-N) relationships in OLAP cubes can be tricky, as they can introduce ambiguity and potential double counting of facts.
+
+For instance, while examining beneficiary demographics, a potential N-to-N relationship is the "tags". We considered a star scheme that looks like:
+
+```
+beneficiaryDemographics {
+  facts: [{
+    gender: String,
+    age: Int,
+    tagId: Int,
+    count: Int
+  }, ...],
+  dimensions: {
+    tag: [{id: Int, name: String}, ...],
+  }
+}
+```
+
+However, grouping by gender or age can cause double counting due to lost relational information among tags.
+
+To tackle this, we chose a non-standard star scheme. Since our data isn't restricted to relational table layouts, dimensions can be represented by arrays. This leads to:
+
+```
+beneficiaryDemographics {
+  facts: [{
+    gender: String,
+    age: Int,
+    tagIds: Int[],
+    count: Int
+  }, ...],
+  dimensions: {
+    tag: [{id: 1, name: ....}, ...],
+  }
+}
+```
+
+### B. Data Transformation
+
+The BE has to transform Boxtribute's data to be able to return the data in the desired fromat from above. Now, the question arises when and how to do these transformations. We see the following options:
+
+#### B.1 On-the-fly Transformation During GraphQL Requests
+
+This is straight-forward solution and how we started for the first visualizations - Directly transform operational data from Boxtribute in response to incoming graphQL requests.
+
+**Pros**:
+
+- you only need to write the query
+- live data
+
+**Cons**:
+
+- query with many reads directly from the Boxtribute db
+
+  --> increase of cost & latency, not scalable, interference with the Boxtribute db
+
+#### B.2 Scheduled Nightly Transformation
+
+Since the data returned for analytics does not have to be live, we have the option to transform the data nightly and cache it.
+
+**Caching options**: Database, File-System, Redis or similar.
+
+**Cache Update Strategies**:
+
+- _Full Refresh_: Purge the cache nightly and transform all data anew.
+- _Incremental Update_: Sequentially extend the cache with data changes that occurred during the day.
+
+**Pros**:
+
+- clear separation of operational data in Boxtribute db and analytical data in a cache
+
+  --> separation of high risk data, no interference with Boxtribute
+
+- improved queries with few reads for the public endpoint
+
+  --> no huge increase in cost (depending on the cache), low latency
+
+**Cons**:
+
+- no live data
+- more code writing nightly transformation and caching
+
+#### B.3 Reactive Transformation on Boxtribute Data Updates
+
+Instead of preparing the data for analytics once a night, we could also directly update the cache when a change happens to the Boxtribute operational db. We have not really considered this option, but it would be possible.
```

Comment:
This would imply either
- implementing the cache update logic in both dropapp and v2, or
- use implementation on DB level
First one is double work; second one is too raw, and also inconsistent to have logic on DB level in my opinion. All in all B.3 is a no-go in my opinion